File: emcrypt-core/src/main/java/com/beam/emcryptcore/model/keyman/mail/Content.java
package com.beam.emcryptcore.model.keyman.mail;

import com.beam.emcryptcore.base.Base;
import com.beam.emcryptcore.model.common.Language;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;
import org.springframework.data.annotation.Transient;
import org.springframework.data.annotation.TypeAlias;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

@Document("Content")
@TypeAlias("Content")
@AllArgsConstructor
@NoArgsConstructor
@Accessors(chain = true)
@Data
@Builder
public class Content extends Base {

    private Type type;
    private Map<Language, String> html;

    @Transient
    private boolean missing;

    @Override
    public Content newIdAndCreated() {
        super.newIdAndCreated();
        missing = true;
        html = new HashMap<>();
        Arrays.stream(Language.values()).forEach(l -> html.put(l, null));
        return this;
    }
}

File: emcrypt-web/src/main/ui/src/app/modules/company/modules/settings/model/content.ts
import { Base, Language } from "src/app/common/models/model";

export enum Type {
  REGULAR,
}

export class Content extends Base {
  missing: boolean;
  selected: boolean;
  type: Type;
  html: any;

  init(_content: Content) {
    super.init(_content);

    this.missing = _content.missing;
    this.type = _content.type;
    if (!_content.missing) {
      this.html = _content.html;
    } else {
      this.html = {};
    }
  }
}
